File: apps/chat/libs/chat-domain/src/main/java/com/example/chat/domain/friendship/FriendshipId.java
package com.example.chat.domain.friendship;

import org.springframework.lang.NonNull;

import java.util.UUID;

/**
 * 친구 관계 ID (Value Object)
 */
public record FriendshipId(String value) {
    public FriendshipId {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("FriendshipId cannot be blank");
        }
    }

    public static FriendshipId of(String value) {
        return new FriendshipId(value);
    }

    public static FriendshipId generate() {
        return new FriendshipId(UUID.randomUUID().toString());
    }
}

File: apps/chat/libs/chat-domain/src/main/kotlin/com/example/chat/domain/friendship/FriendshipId.kt
package com.example.chat.domain.friendship

import java.util.*

/**
 * 친구 관계 ID (Value Object)
 */
@JvmInline
value class FriendshipId(val value: String) {
	init {
		require(value.isNotBlank()) { "FriendshipId cannot be blank" }
	}

	companion object {
		/**
		 * 기존 ID로부터 생성
		 */
		fun of(value: String) = FriendshipId(value)

		/**
		 * 새로운 ID 생성
		 */
		fun generate() = FriendshipId(UUID.randomUUID().toString())
	}

	override fun toString(): String = value
}
